packages/braid-design-system/src/lib/components/Button/Button.tsx

```diff
@@ -355,21 +357,42 @@ export const ButtonText = ({
         size={size}
         baseline={false}
       >
-        {icon ? (
+        {icon && iconPosition === 'leading' ? (
           <Box
             component="span"
             paddingRight="xsmall"
-            className={
+            className={[
+              styles.nowrap,
               shouldReducePaddingX || bleed
                 ? null
-                : negativeMargin('left', 'xxsmall')
-            }
+                : negativeMargin('left', 'xxsmall'),
+            ]}
           >
             {icon}
+            {
+              '⁠' /* Word joiner character, a zero-width non-breaking character to prevent the icon wrapping onto its own line */
+            }
           </Box>
         ) : null}
         {children}
         {loading ? <ButtonLoader /> : null}
+        {!loading && icon && iconPosition === 'trailing' ? (
+          <Box
+            component="span"
+            paddingLeft="xsmall"
+            className={[
+              styles.nowrap,
+              shouldReducePaddingX || bleed
+                ? null
+                : negativeMargin('right', 'xxsmall'),
+            ]}
+          >
+            {
+              '⁠' /* Word joiner character, a zero-width non-breaking character to prevent the icon wrapping onto its own line */
+            }
```

Comment:
```suggestion
            {/* Word joiner character, a zero-width non-breaking character to prevent the icon wrapping onto its own line */}
            &#8288;
```

This makes it easier to [look it up] on the web ([see also]).

[look it up]: https://www.fileformat.info/info/unicode/char/2060/index.htm
[see also]: https://stackoverflow.com/a/28405917

Reply:
Nice one yeah, i'll add this to the shared component i created for this problem.